src/lsptoolshost/roslynLanguageServer.ts

```diff
@@ -452,6 +452,16 @@ export class RoslynLanguageServer {
         // Save user's DOTNET_ROOT env-var value so server can recover the user setting when needed
         env.DOTNET_ROOT_USER = process.env.DOTNET_ROOT ?? 'EMPTY';
 
+        if (languageServerOptions.crashDumpPath) {
+            // Enable dump collection
+            env.DOTNET_DbgEnableMiniDump = '1';
+            // Collect heap dump
+            env.DOTNET_DbgMiniDumpType = '2';
+            // Collect crashreport.json with additional thread and stack frame information.
+            env.DOTNET_EnableCrashReport = '1';
+            env.DOTNET_DbgMiniDumpName = path.join(languageServerOptions.crashDumpPath, '%e.%d.dmp');
```

Comment:
You might want to comment what the substitutions do here, especially because I don't see what %d does in the docs. Did you mean %p?

Reply:
Ah %d is the old one, see https://github.com/dotnet/runtime/blob/main/docs/design/coreclr/botr/xplat-minidump-generation.md#configurationpolicy

I'll add a comment and change to %p